driver/driver.feature

```diff
@@ -0,0 +1,1265 @@
+    Then transaction is open: false
+    Then connection open read transaction for database: does-not-exist; fails
+    Then transaction is open: false
+
+
+  Scenario: Driver can open and close transactions of different types
+    Then transaction is open: false
+    When connection open schema transaction for database: typedb
+    Then transaction has type: schema
+    Then transaction is open: true
+    When transaction closes
+    Then transaction is open: false
+
+    When connection open write transaction for database: typedb
+    Then transaction has type: write
+    Then transaction is open: true
+    When transaction closes
+    Then transaction is open: false
+
+    When connection open read transaction for database: typedb
+    Then transaction has type: read
+    Then transaction is open: true
+    When transaction closes
+    Then transaction is open: false
+
+
+  Scenario: Driver can commit transactions of schema and write types, cannot commit transaction of type read
+    When connection open schema transaction for database: typedb
+    Then transaction has type: schema
+    Then transaction is open: true
+    When transaction commits
+    Then transaction is open: false
+
+    When connection open write transaction for database: typedb
+    Then transaction has type: write
+    Then transaction is open: true
+    When transaction commits
+    Then transaction is open: false
+
+    When connection open read transaction for database: typedb
+    Then transaction has type: read
+    Then transaction is open: true
+    Then transaction commits; fails
+    Then transaction is open: false
+
+
+  # TODO: Fix rollbacks on the server side
+#  Scenario: Driver can rollback transactions of schema and write types, cannot rollback transaction of type read
+#    When connection open schema transaction for database: typedb
+#    Then transaction has type: schema
+#    Then transaction is open: true
+#    When transaction rollbacks
+#    Then transaction is open: false
+#
+#    When connection open write transaction for database: typedb
+#    Then transaction has type: write
+#    Then transaction is open: true
+#    When transaction rollbacks
+#    Then transaction is open: false
+#
+#    When connection open read transaction for database: typedb
+#    Then transaction has type: read
+#    Then transaction is open: true
+#    Then transaction rollbacks; fails
+#    Then transaction is open: false
+
+
+  # TODO: Check options setting and retrieval
+
+
+  # TODO: Fix on_close
+#  Scenario: Driver can schedule "transaction on close" jobs
+#    Given connection does not have database: created-after-schema
+#    Given connection does not have database: created-after-write
+#    Given connection does not have database: created-after-read
+#    When connection open schema transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-schema
```

Comment:
Is another strange step. Wanted to make something interesting with the possibility to check its execution through a common step.

Reply:
hmm ok, im not so worried about testing this as a primary feature yet

Reply:
It's something we implement for the drivers, it's a part of the API, so it should be tested. If we are not ready to give it to the users, we should just comment it out in production or remove it completely.

Reply:
There are no more "primary" features for the drivers to test or not to test. There is a short list of features that we want to always test.
If we want something extra for a specific driver, try to cover it by integration tests, you may also use the example test for these features. If not, use BDDs and reimplement it everywhere.

Reply:
Then we should fix these now :)